content/_code-samples/escrow/js/create-escrow.js

```diff
@@ -1,57 +1,57 @@
 'use strict'
-const RippleAPI = require('ripple-lib').RippleAPI
-const cc = require('five-bells-condition')
-const crypto = require('crypto')
-
-const myAddr = 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn'
-const mySecret = 's████████████████████████████'
-
-// Construct condition and fulfillment
-const preimageData = crypto.randomBytes(32)
-const myFulfillment = new cc.PreimageSha256()
-myFulfillment.setPreimage(preimageData)
-const conditionHex = myFulfillment.getConditionBinary().toString('hex').toUpperCase()
-
-console.log('Condition:', conditionHex)
-console.log('Fulfillment:', myFulfillment.serializeBinary().toString('hex').toUpperCase())
-
-// Construct transaction
-const currentTime = new Date()
-const myEscrow = {
-  "destination": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn", // Destination can be same as source
-  "destinationTag": 2017,
-  "amount": "0.1113", //decimal XRP
-  "condition": conditionHex,
-  "allowExecuteAfter": currentTime.toISOString() // can be executed right away if the condition is met
-}
-const myInstructions = {
-  maxLedgerVersionOffset: 5
-}
+if (typeof module !== "undefined") {
+  // Use var here because const/let are block-scoped to the if statement.
+  var xrpl = require('xrpl');
+};
+
+const cc = require('five-bells-condition');
+const crypto = require('crypto');
+
+const main = async () => {
+  try {
+
+    // Construct condition and fulfillment ---------------------------------------
+    const preimageData = crypto.randomBytes(32);
+    const myFulfillment = new cc.PreimageSha256();
+    myFulfillment.setPreimage(preimageData);
+    const conditionHex = myFulfillment.getConditionBinary().toString('hex').toUpperCase();
+
+    console.log('Condition:', conditionHex);
+    console.log('Fulfillment:', myFulfillment.serializeBinary().toString('hex').toUpperCase());
+
+    // Connect -------------------------------------------------------------------
+    const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233');
+    await client.connect();
+
+    // Get credentials from the Testnet Faucet ------------------------------------
+    console.log("Requesting an address from the Testnet faucet...");
+    const { wallet } = await client.fundWallet();
+    console.log("Wallet: ", wallet.address);
+
+    const firstRippleEpoch = 946684800;
+    const escrowCreateTransaction = {
+      "TransactionType": "EscrowCreate",
+      "Account": wallet.address,
+      "Destination": wallet.address,
+      "Amount": "6000000", //drops XRP
+      "DestinationTag": 2023,
+      "Condition": conditionHex,
+      "Fee": "12",
+      "FinishAfter": Math.floor((new Date().getTime() / 1000) + 120) - firstRippleEpoch, // 2 minutes from now
```

Comment:
You can use `isoTimeToRippleTime` from xrpl.js to simplify this line - https://js.xrpl.org/functions/isoTimeToRippleTime.html

Reply:
It might also be worth adding a print statement explaining that "This escrow can be finished 2 minutes from now." so if they want to play with it they can interpret the hard-to-understand FinishAfter number.

Reply:
Added time calculations and console logs